usr/share/rear/lib/opal-functions.sh

```diff
@@ -26,6 +26,22 @@ function opal_devices() {
     sedutil-cli --scan | awk '$1 ~ /\/dev\// && $2 ~ /2/ { print $1; }'
 }
 
+function opal_device_disks() {
+    local device="${1:?}"
+    # prints all block devices belonging to the given Opal device.
+    # Normally, this is just the Opal device itself, however, NVME devices have one or more namespaces per primary
+    # device and these namespaces act as disks.
+
+    case "$device" in
+        (*/nvme*)
+            echo "$device"n[0-9]  # consider all namespace block devices (NOTE: relies on nullglob)
```

Comment:
`"$device"n[0-9]` matches only up to namespace number 9 i.e. `/dev/nvme...n9`
so e.g. `/dev/nvme0n10` and higher namespace numbers are ignored.

To be more on the safe side I suggest
```
# consider all namespace block devices up to /dev/nvme...n99 (NOTE: relies on nullglob)
echo "$device"n[0-9] "$device"n[0-9][0-9]
```

I think more than 10 namespaces might happen in practice
while more than 100 namespaces never happen in real world.

But the latter is probably only wishful thinking because
https://nvmexpress.org/wp-content/uploads/NVM-Express-1_4-2019.06.10-Ratified.pdf
reads (excerpts)
```
... namespace ID (NSID) ...

6.1.2 Valid and Invalid NSIDs
...
Any NSID is valid, except if that NSID is 0h or greater than the
Number of Namespaces field reported in the Identify Controller
data structure (refer to Figure 247).
NSID FFFFFFFFh is a broadcast value that is used to specify
all namespaces. 
```
so it seems any number up to FFFFFFFFh could appear in real world.

Reply:
The next question is how namespaces > 9 block device nodes are shown
as decimal numbers like /dev/nvme0n12
or as hex numbers like /dev/nvme0nc or /dev/nvme0nC or /dev/nvme0n0c

Reply:
So far, I have not seen namespace IDs beyond 1 being used in the wild. But I don't mind if we extend the scheme.

The [kernel](https://github.com/torvalds/linux/blob/e9287bd248160880ff9476eea705fa47f75c6136/drivers/nvme/host/nvme.h#L711) will use decimal digits:
```C
	sprintf(disk_name, "nvme%dn%d", ctrl->instance, ns->head->instance);
```